apps/wallet-mobile/src/features/Links/useCases/RequestedAdaPaymentWithLinkScreen/RequestedAdaPaymentWithLinkScreen.tsx

```diff
@@ -0,0 +1,99 @@
+import {LinksTransferRequestAdaWithLinkParams, useLinks} from '@yoroi/links'
+import {useTheme} from '@yoroi/theme'
+import * as React from 'react'
+import {ScrollView, StyleSheet, Text, View, ViewProps} from 'react-native'
+import {SafeAreaView} from 'react-native-safe-area-context'
+
+import {Button, Spacer, useModal} from '../../../../components'
+import {isEmptyString} from '../../../../utils/utils'
+import {useStrings} from '../../common/useStrings'
+import {ShowDisclaimer} from './ShowDisclaimer/ShowDisclaimer'
+
+export const RequestedAdaPaymentWithLinkScreen = ({
+  params,
+  isTrusted,
+  onContinue,
+}: {
+  params: LinksTransferRequestAdaWithLinkParams
+  isTrusted?: boolean
+  onContinue: () => void
+}) => {
+  const strings = useStrings()
+  const {styles} = useStyles()
+  const {actionFinished} = useLinks()
+  const {closeModal} = useModal()
+
+  // TODO: revisit check with product
+  const disclaimerStyle = isTrusted ? styles.text : styles.text
+  const description = isTrusted
+    ? strings.trustedPaymentRequestedDescription
+    : strings.untrustedPaymentRequestedDescription
+
+  const handleOnCancel = () => {
+    actionFinished()
+    closeModal()
+  }
+
+  return (
+    <SafeAreaView edges={['bottom', 'left', 'right']} style={styles.root}>
+      <ScrollView bounces={false}>
+        <ShowDisclaimer title={strings.disclaimer}>
+          <Text style={disclaimerStyle}>{description}</Text>
+        </ShowDisclaimer>
+
+        <Spacer height={16} />
+
+        {/* TODO: revisit SHOW the app name or unknown */}
```

Comment:
yap, all copies need to be checked with Product and there are some special states that need errors